### DISCUSSION TITLE
DSS Color and noise

### DISCUSSION THREAD

**Discussion by rnclark-psi** [deepskystacker/DSS#144] [General]
DSS is a great program and used by many people and I thank you for your efforts.

I have been analyzing various methods amateur astrophotographers use to process their data.  I have some information about DSS.  Please see https://clarkvision.com/articles/sensor-calibration-and-color/

Two things that would greatly improve DSS would be a better demosaicing algorithm (see Figures 1-5 10, 11a-c, 12a-c, 13a-c) which all show results from DSS.  The second thing would be a button where a user could input the 9 values for doing a color matrix correction.  That would greatly improve color output (probably also need multipliers for daylight withe balance).

**Comment by perdrix52** [comment]
Demosaicing of Camera Raw files is done by libraw and the white balance coefficients from the camera are provided to DSS by libraw.

You website shows image stacking using DSS 3.3.2 - which is a VERY old release.   Have you tried the recent releases (e.g. 4.2.6)

![image](https://user-images.githubusercontent.com/20211762/182038237-63887809-4a93-4e3b-93cb-a152967f61f8.png)


I can indeed see that the image as processed by libraw does appear different from that shown by Adobe Camera Raw even though DSS was set to "Use Camera White Balance" and ACR set to use "As shot".   All I did in DSS to view the image better was to move the gamma slider at the top right of the DSS window to the left.   My feeling is that ACR has applied some form of "automatic" vibrance adjustement even though I didn't ask for it.

You mention Rawtherapee, but that is a stand-alone programme, not a library for processing raw files.   AFAIK, libraw is the only free raw converter library in the game.   If you know of a better library for doing Raw file processing, I am "all ears"!

**Reply by rnclark-psi** [reply]
Yes, rawtherapee is an independent program, but open source.  It is coded many of the raw converter algorithms.  I typically use the LMMSE algorithm.  You may be able to use the rawthreapee code (check license).  Also, I believe rawtherapee uses libraw too.

**Comment by rnclark-psi** [comment]
On 7/31/22 10:54, David C. Partridge wrote:
>
> Demosaicing of Camera Raw files is done by libraw and the white 
> balance coefficients from the camera are provided to DSS by libraw.
>
> You website shows image stacking using DSS 3.3.2 - which is a VERY old 
> release. Have you tried the recent releases?
>
>
Hello David,

Thank you for responding.

I'm using DSS 4.2.6 for the sensor calibration article.

I think libraw provides generic white balance but not a camera specific 
white balance.

Roger

**Comment by perdrix52** [comment]
AFAIK libraw provides the WB values specific to the Camera for Daylight and the actual WB values if using “Use Camera White Balance” is set.

 

I’m loath to approach the libraw maintainers as the last time I tried to discuss an issue with them they just treated me like something you wipe off your boot.

 

If you wish to suggest to them that their decoding could be improved, please go ahead 😊  You’re a braver man than I!

 

David

**Comment by perdrix52** [comment]
I found this: [https://www.libraw.org/node/2478](https://www.libraw.org/node/2478) which give me hope that I can grab the G50 illuminant colour correction matrix and apply that after WB processing.

**Reply by perdrix52** [reply]
But sadly the matrix libraw returns in most cases is all zeros :(

I'll go digging in the RawTherapee code to see it they have any CCM data.

D.

**Comment by perdrix52** [comment]
Roger,

I'm back again, do you know where I might find a downloadable set of CCMs for "most" cameras (for "daylight WB" by which I will have to assume to be D50 given I haven't ever seen one for D65)
